### ISSUE TITLE
Warning before completely removing your data/roles

### ISSUE THREAD

**Issue by MoonlightCapital** [Dougley/MBv2#107]
**Is your feature request related to a problem? Please describe.**
As version 5.0.0, MegaBot will remove all the exp and the roles earned by an user if they're inactive for over a month. This may surprise a few users if they are unaware of the change.

**Describe the solution you'd like**
A week (or something) before the complete removal, MegaBot would DM the user alerting them their exp/ranks are at risk.

**Additional context**
No gnoming in this suggestion.

**Comment by Zacatero** [comment]
I dislike this. I feel like it goes against the point of even having the timeout in the first place, If the point is to weed out inactive custodians, why should we alert them before it happens so they can do one thing and then go back to being inactive.

**Comment by Dougley** [comment]
I agree with @Zacatero on this, alerting might cause them to start being active again momentarily only to keep their stats. In edge cases where deletion takes people completely by surprise due to circumstances we as mods might decide to reinstate them, on our own discretion.

**Comment by MoonlightCapital** [comment]
What about those who haven't read your announcement and are completely unaware of the change? I feel like the complete reset of stats is for people who haven't used discord completely to purge the DB from them.

**Comment by Dougley** [comment]
> 
> 
> What about those who haven't read your announcement and are completely unaware of the change? 

You'd be inactive then, right?

**Comment by MoonlightCapital** [comment]
Not reading announcements is not equal to inactive. If you don't like a pre-emptive warning, at least make the bot DM an user when their stats get reset.

**Comment by Zacatero** [comment]
> Not reading announcements is not equal to inactive.

It kind of is. Custodians should be reading the announcements. If it takes you more than a month to read the announcements, that's pretty inactive. But I do agree that there's no issue with having a message after the fact like "Hey you were inactive so your roles are gone" or whatever.

**Comment by Kitty-Therapist** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Warning before completely wiping out your roles and XP](https://trello.com/c/ErOieO2E/9-warning-before-completely-wiping-out-your-roles-and-xp)
